noir-projects/noir-protocol-circuits/crates/types/src/traits.nr

```diff
@@ -55,6 +56,11 @@ impl ToField for U128 {
         self.to_integer()
     }
 }
+impl<N> ToField for str<N> {
+    fn to_field(self) -> Field {
+        field_from_bytes(self.as_bytes(), true)
```

Comment:
Right, that's a good one. However this impl is finally not used in favor of `serialize` in this PR.

I'm gonna add the assertion and keep it since we could probably remove a few instances of `FieldCompressedString` in favor of this simpler trait.